### ISSUE TITLE
crash on apply profile 

### ISSUE THREAD

**Issue by amarchiguess** [terrymacdonald/DisplayMagician#139]
When double clicking a profile on desktop, display magician starts and crash. it's not applied. Profile was created with the same version. It looks like an array is empty or not finding somethign referenced by the link. I attach the log

[DisplayMagician.log](https://github.com/terrymacdonald/DisplayMagician/files/9435288/DisplayMagician.log)

**Comment by terrymacdonald** [comment]
Hi,

When you say you click on a profile on the desktop, are you right clicking
on the desktop background and using the pop up menu to select the display
profile? Or are you saving a desktop shortcut to the display profile and
double-clicking on the desktop shortcut to change the profile?


Can you please create a Support ZIP File and upload it to this issue? You
can do that by clicking on the Settings button on the main DisplayMagician
screen, then clicking on the Create Support ZIP File button down in the
Settings window.

Please then attach the zip file created to this issue using your web
browser.

This should then give me all the config files I need to see what's
happening, in addition to the logs.

Thanks
Terry

**Comment by amarchiguess** [comment]
I'm double clicking on the desktop shortcut. 

The crash was happening when trying to activate "Tre schermi" coming from "Surround", after playing Assetto Corsa Competizione and Dirt Rally 2 (maybe it helps to know, maybe not).

I did a few more tests right now. 
I re-configured the 3 separate screens with nVidia control panel, and saved a new profile named "Tre schermi 2". 
I then tried to switch again to Surround and then back to the 'broken' "Tre schermi", and now it works. So there is something weird going on that is  happening only in particular scenarios, maybe after running a game.

I attach the zip file
[DisplayMagician-Support-20220826-2149.zip](https://github.com/terrymacdonald/DisplayMagician/files/9436075/DisplayMagician-Support-20220826-2149.zip)

**Comment by terrymacdonald** [comment]
Ok, thanks for that I'll take as look later today and see what I can find.

Oh, what version of windows do you use? And what version of NVIDIA driver
are you running? What make and model of screens are you running and how are
they connected? If you have more than one video card, which display is
connected to which video card?

And lately, have you upgraded from windows 10 to windows 11 recently? Were
the offending display profiles created in windows 10?

Thanks
Terry

**Comment by amarchiguess** [comment]
Windows 10 build 19044.1889
nVidia drivers 516.94 (latest)
3 monitors Asus VG27AQ1A (2 with display 1 port, 1 with hdmi, or the other way around, 2 with hdmi, 1 with display port, I never remember which is which)
Asus RTX 2080 TI
No udates from windows 10 to 11, same old windows 10.

**Comment by terrymacdonald** [comment]
Right, I've had a look through your logs and config, and it looks as though for some reason, NVIDIA isn't happy with the Saved Topology, and it believes the topology in 'Tre schermi' isn't valid:

```
2022-08-26 19:43:02.6721|ERROR|DisplayMagician.Program|NVIDIALibrary/SetActiveConfig: The topology passed in is not currently possible. NvAPI_Mosaic_SetDisplayGrids() returned error code NVAPI_TOPO_NOT_POSSIBLE|
2022-08-26 19:43:02.6721|TRACE|DisplayMagician.Program|ProfileItem/SetActive: The NVIDIA display settings within profile Tre schermi were NOT applied correctly.|
2022-08-26 19:43:02.6721|ERROR|DisplayMagician.Program|ProfileRepository/ApplyProfile: Error applying the NVIDIA Profile!|
```

One of the reasons that DisplayMagician works so well typically is that it is simply an NVIDIA/AMD config recorder. It literally takes a snapshot of the current NVIDIA configuration, and stores them for use later on. This 'low touch' way of operating is typically very reliable, but it also means that DisplayMagician relies on the NVIDIA configuration never changing.

Every so often, something is changed in the NVIDIA world that affects the DisplayMagician config. Sometimes, NVIDIA changes the values of the data it returns. Other times, the NVIDIA driver internal database is updated through some display changes, or its reinstalled, or some changes are made to the NVIDIA driver that render the NVIDIA config stored in DisplayMagician as invalid. In your case, I think this is what happened.

The only way to properly fix this in DisplayMagician, is for you to:

- Delete the old Tre Schermi display profile
- Create a new Tre Schermi display profile
- Go through all your Game Shortcuts that used the old Tre Schermi display profile, and update them to use the new Tre Schermi display profile.

Then it should all work again as it did before. Can you please do the above if you haven't already, and please reply back with whether it worked.

Unfortunately you've hit one of the design tradeoffs that I had to make when designing DisplayMagician - recording settings directly from the NVIDIA driver means we are affected if any of those settings are considered 'invalid' by the NVIDIA driver at a later date. Luckily the fix is pretty straightforward.

Hope that helps!

Thanks
Terry

**Comment by amarchiguess** [comment]
I do not know if this is the case. After I created a second "Tre schermi", now also the one that was broken works again. I just did a test an hour ago. "Surround"->"Tre schermi"->"Surround"->Start ACC->"Tre Schermi". Everything worked like a charm.
So there must be some temporary and not predictable (at the moment) nVidia issue. I will check when it happen next time. 
Thank you for your quick feedback!
